### PR TITLE
feat: add logging + cleanup/refactor for coverage

### PR DESCRIPTION
**Pull request by cameronmeissner**

### CODE REVIEW

**Review by AlisonB319 — APPROVED** [review]

File: cmd/client/main.go [hunk 1]
```diff
@@ -9,53 +9,21 @@ import (
...
-			false,
-			"enable debug logging (WILL LOG AUTHENTICATION DATA)",
-		)
+		clientID  = flag.String("client-id", "", "The client ID for the assigned identity to use.")
+		logFormat = flag.String("log-format", "json", "Log format: json or text, default: json")
+		nextProto = flag.String("next-proto", "", "ALPN Next Protocol value to send.")
+		debug     = flag.Bool("debug", false, "enable debug logging (WILL LOG AUTHENTICATION DATA)")
 	)
 
 	flag.Parse()
```
Comment (cameronmeissner) [inline cmd/client/main.go:25]: I think I'll move us over to cobra in a separate PR

File: pkg/client/client.go [hunk 2]
```diff
@@ -90,71 +101,95 @@ func (c *tlsBootstrapClientImpl) GetBootstrapToken(ctx context.Context) (string,
...
+		},
+	}, nil
+}
 
+func loadExecCredential() (*datamodel.ExecCredential, error) {
+	execInfo := os.Getenv(kubernetesExecInfoVarName)
+	if execInfo == "" {
+		return nil, fmt.Errorf("%s must be set to retrieve bootstrap token", kubernetesExecInfoVarName)
+	}
 	execCredential := &datamodel.ExecCredential{}
```
Comment (juan-lee) [inline pkg/client/client.go:189]: prefer allocating on the stack vs. heap where possible. i.e., var `execCredential datamodel.ExecCredential`

File: pkg/client/client.go [hunk 3]
```diff
@@ -90,71 +101,95 @@ func (c *tlsBootstrapClientImpl) GetBootstrapToken(ctx context.Context) (string,
 			}),
 		}),
 	)
+	c.logger.Info("dialed TLS bootstrap server and created GRPC connection")
+
+	return conn, nil
+}
+
+func (c *tlsBootstrapClientImpl) GetBootstrapToken(ctx context.Context) (string, error) {
+	conn, err := c.setupClientConnection(ctx)
```
Comment (juan-lee) [inline pkg/client/client.go:110]: Does `conn` need to be cleaned up?

File: pkg/client/auth.go [hunk 4]
```diff
@@ -12,50 +12,57 @@ import (
...
+		var clientID string
+		if azureConfig.UserAssignedIdentityID != "" {
+			clientID = azureConfig.UserAssignedIdentityID
+		}
+		tokenResponse, err := c.imdsClient.GetMSIToken(ctx, baseImdsURL, clientID)
+		if err != nil {
+			return "", fmt.Errorf("unable to get MSI token from IMDS: %w", err)
+		}
+		return tokenResponse.AccessToken, nil
+	} else {
```
Comment (juan-lee) [inline pkg/client/auth.go:47]: is else necessary?

File: pkg/client/client.go (hunk 3 above)
Comment (cameronmeissner) [reply pkg/client/client.go:110]: good catch, thanks

File: pkg/client/auth.go (hunk 4 above)
Comment (cameronmeissner) [reply pkg/client/auth.go]: norp, removed

**Pull request merged**